puppet/lib/puppet/indirector/catalog/puppetdb.rb

```diff
@@ -54,6 +54,7 @@ def munge_catalog(catalog, producer_timestamp, extra_request_data = {})
       end
 
       add_code_id_if_missing(data)
+      add_job_id_if_missing(data)
```

Comment:
I think this actually needs to pull the job_id from the extra_request_data, like transaction_uuid. The Puppet catalog compilation code isn't putting it into the catalog itself.